Clarify set_page_config docstring and exception message

MathCatsAnd opened the pull request:
## 📚 Context
Added documentation and exception clarification per Issue #6572

## 🧠 Description of Changes
Updated a few words in the `set_page_config` docstring.
Fixed associated exception message and tests to reflect being restricted to one call per page instead of per (whole) app.

**Contribution License Agreement**
By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

MathCatsAnd committed:
Clarify set_page_config documentation

Added clarification per Issue #6572

willhuang1997 reviewed (approved):
Thanks for the contribution! Love what you do on the streamlit forum and appreciate it so much!

snehankekre reviewed (approved):
Thank you @MathCatsAnd!! The change looks good to me. 

I was thinking we could also amend the related `StreamlitAPIException` message in the same PR (and edit the PR title+description). WDYT @willhuang1997?

https://github.com/streamlit/streamlit/blob/0122abc5f8de42f4770cec12de0f303ea0337e18/lib/streamlit/runtime/scriptrunner/script_run_context.py#L91 

The above can be changed to 

```suggestion
"`set_page_config()` can only be called once per app page, " 
```

Similarly, these two lines in the related e2e test can be updated to add `page` to the end:

- https://github.com/streamlit/streamlit/blob/0122abc5f8de42f4770cec12de0f303ea0337e18/e2e/specs/st_set_page_config.spec.js#L74
- https://github.com/streamlit/streamlit/blob/0122abc5f8de42f4770cec12de0f303ea0337e18/e2e/specs/st_set_page_config.spec.js#L84

MathCatsAnd committed:
Merge branch 'streamlit:develop' into page-config-documentation

MathCatsAnd committed:
Updated exceptions/tests for #6572

Updated exceptions and tests to clearly reflect the restriction per page instead of per whole app, consistent with documentation tweak in #6572

MathCatsAnd committed:
Merge branch 'page-config-documentation' of https://github.com/MathCatsAnd/streamlit into page-config-documentation

MathCatsAnd commented:
Thanks @snehankekre. I wasn't sure where all the associated exceptions and tests were. I've added the updates you mentioned to the PR.

snehankekre commented:
Woohoo, thanks for updating the exception and tests, @MathCatsAnd ! 😄

MathCatsAnd commented:
I've updated the top comment as well @snehankekre. Feel free to update the name of the commit as desired. I'm not 100% confident in my naming sense, yet.

If I were to guess, maybe:
`Fix set_page_config documentation & exception`
